checks/evaluation/ci_tests.go

```diff
@@ -16,122 +16,62 @@ package evaluation
-		prCheckSuccessful, err := prHasSuccessfulCheck(r, dl)
-		if err != nil {
-			return checker.CreateRuntimeErrorResult(CheckCITests, err)
-		}
-		if prCheckSuccessful {
-			totalTested++
-			foundCI = true
-		}
+const CheckCITests = "CI-Tests"
 
-		if !foundCI {
-			// Log message says commit, but really we only care about PRs, and
-			// use only one commit (branch HEAD) to refer to all commits in a PR
+func CITests(name string,
+	findings []finding.Finding,
+	dl checker.DetailLogger,
+) checker.CheckResult {
+	expectedProbes := []string{
+		testsRunInCI.Probe,
+	}
+	if !finding.UniqueProbesEqual(findings, expectedProbes) {
+		e := sce.WithMessage(sce.ErrScorecardInternal, "invalid probe results")
+		return checker.CreateRuntimeErrorResult(name, e)
+	}
 
+	// Debug PRs that were merged without CI tests
+	for i := range findings {
+		f := &findings[i]
+		if f.Outcome == finding.OutcomeNegative {
 			dl.Debug(&checker.LogMessage{
-				Text: fmt.Sprintf("merged PR %d without CI test at HEAD: %s", r.PullRequestNumber, r.HeadSHA),
+				Text: f.Message,
 			})
 		}
 	}
 
-	if totalMerged == 0 {
+	// check that the project has pull requests
+	if noPullRequestsFound(findings) {
 		return checker.CreateInconclusiveResult(CheckCITests, "no pull request found")
 	}
 
-	reason := fmt.Sprintf("%d out of %d merged PRs checked by a CI test", totalTested, totalMerged)
-	return checker.CreateProportionalScoreResult(CheckCITests, reason, totalTested, totalMerged)
-}
+	totalMerged := findings[0].Values["totalMerged"]
+	totalTested := findings[0].Values["totalTested"]
 
-// PR has a status marked 'success' and a CI-related context.
-//
-//nolint:unparam
-func prHasSuccessStatus(r checker.RevisionCIInfo, dl checker.DetailLogger) (bool, error) {
-	for _, status := range r.Statuses {
-		if status.State != success {
-			continue
-		}
-		if isTest(status.Context) || isTest(status.TargetURL) {
-			dl.Debug(&checker.LogMessage{
-				Path: status.URL,
-				Type: finding.FileTypeURL,
-				Text: fmt.Sprintf("CI test found: pr: %s, context: %s", r.HeadSHA,
-					status.Context),
-			})
-			return true, nil
-		}
+	if totalMerged < totalTested || len(findings) < totalTested {
+		e := sce.WithMessage(sce.ErrScorecardInternal, "invalid finding values")
+		return checker.CreateRuntimeErrorResult(name, e)
 	}
-	return false, nil
-}
 
-// PR has a successful CI-related check.
-//
-//nolint:unparam
-func prHasSuccessfulCheck(r checker.RevisionCIInfo, dl checker.DetailLogger) (bool, error) {
-	for _, cr := range r.CheckRuns {
-		if cr.Status != "completed" {
-			continue
-		}
-		if cr.Conclusion != success {
-			continue
-		}
-		if isTest(cr.App.Slug) {
-			dl.Debug(&checker.LogMessage{
-				Path: cr.URL,
-				Type: finding.FileTypeURL,
-				Text: fmt.Sprintf("CI test found: pr: %d, context: %s", r.PullRequestNumber,
-					cr.App.Slug),
-			})
-			return true, nil
-		}
-	}
-	return false, nil
-}
+	// Log all findings
+	checker.LogFindings(nonNegativeFindings(findings), dl)
```

Comment:
weren't these messages debug as well under the old code?

Reply:
Yes, in the old evaluation code, if a PR does not have successful check or status, Scorecard would log it at the Debug level:

https://github.com/ossf/scorecard/blob/5dc03b72e109d67afe4c91a0fbe52a3c1678a97f/checks/evaluation/ci_tests.go#L65-L67

In the new, modified evaluation code, we create an `OutcomeNegative` if PR does not have successful check or status, and we log that at the debug level here:
https://github.com/ossf/scorecard/blob/a7970c6231277f2d808da2716c3e28f8a85b6948/checks/evaluation/ci_tests.go#L41-L48

Reply:
I was instead talking about the outcome positives were debug before as well

Reply:
You're right. Changed logging level here: https://github.com/ossf/scorecard/pull/3621/commits/0f6383f0f0c60adc71b8561118929cfb71e84a8a